lib/rucio/core/transfer.py

```diff
@@ -1331,6 +1337,9 @@ def __list_transfer_requests_and_source_replicas(total_workers=0, worker_number=
     if activity:
         sub_requests = sub_requests.filter(models.Request.activity == activity)
 
+    if transfertool:
+        sub_requests = sub_requests.filter(models.Request.transfertool == transfertool)
```

Comment:
You need to make the index hint (Line 1321) conditional here too.
E.g. if no transfertool, use the hint from above, if transfertool, use the new index.